### PR TITLE
[PIRK-36]: Ensure Schema Properties and Registries are Reset at the Beginning of Relevant Tests 

### PR DESCRIPTION
**Pull request by ellisonanne**

### CODE REVIEW

File: src/test/java/test/wideskies/standalone/StandaloneTest.java [hunk 1]
```diff
@@ -49,6 +51,12 @@
 
   public StandaloneTest() throws Exception
   {
+    // Reset the schema properties and registries
+    DataSchemaRegistry.clearRegistry();
+    QuerySchemaRegistry.clearRegistry();
```
Comment (smarthi) [inline src/test/java/test/wideskies/standalone/StandaloneTest.java]: wouldn't it be cleaner if this code was invoked from within tearDown() method?

File: src/test/java/test/wideskies/standalone/StandaloneTest.java [hunk 2]
```diff
@@ -49,6 +51,12 @@
...
+    SystemConfiguration.setProperty("data.schemas", "none");
+    SystemConfiguration.setProperty("query.schemas", "none");
```
Comment (smarthi) [inline src/test/java/test/wideskies/standalone/StandaloneTest.java]: How about moving this code to setUp() or Before() method ?

**Comment by ellisonanne** [comment]
If we are going to move it to a method, I would rather have it in a setup() method so that it is self-contained - i.e., that a test does not rely on previous tests cleaning up after themselves correctly.

**Comment by smarthi** [comment]
What's presently in the Constructor() can be split between methods annotated with @BeforeClass and @AfterClass.  There's no need of a constructor then.

**Comment by ellisonanne** [comment]
Yes - this can be done throughout

**Comment by smarthi** [comment]
I have a fix at https://github.com/smarthi/incubator-pirk/commit/e2c76301b2345ca7371af8b10fedb697182dbf2c

**Comment by ellisonanne** [comment]
Done - pls review

File: src/test/java/test/wideskies/standalone/StandaloneTest.java [hunk 3]
```diff
@@ -45,10 +49,17 @@
...
-  private String stopListFileProp = null;
+  private static String stopListFileProp = null;
 
-  public StandaloneTest() throws Exception
+  @BeforeClass
+  public static void setup() throws Exception
   {
+    // Reset the schema properties and registries
+    DataSchemaRegistry.clearRegistry();
+    QuerySchemaRegistry.clearRegistry();
```
Comment (smarthi) [inline src/test/java/test/wideskies/standalone/StandaloneTest.java]: why r these 2 lines needed here when u already have them in teardown() ?

File: src/test/java/test/general/PartitionUtilsTest.java [hunk 4]
```diff
@@ -39,7 +40,8 @@
 {
   private static final Logger logger = LoggerFactory.getLogger(PartitionUtilsTest.class);
 
-  public PartitionUtilsTest()
+  @BeforeClass
+  public static void setup()
   {
     PrimitiveTypePartitioner primitivePartitioner = new PrimitiveTypePartitioner();
   }
```
Comment (smarthi) [inline src/test/java/test/general/PartitionUtilsTest.java]: This block of code can be removed, this instance of primitivePartitioner is not used in any of the tests.

File: src/test/java/test/wideskies/standalone/StandaloneTest.java (hunk 3 above)
Comment (ellisonanne) [inline src/test/java/test/wideskies/standalone/StandaloneTest.java]: clear before the tests - clean up after the tests

File: src/test/java/test/general/PartitionUtilsTest.java (hunk 4 above)
Comment (ellisonanne) [inline src/test/java/test/general/PartitionUtilsTest.java]: Correct - will remove

**Comment by smarthi** [comment]
Lgtm +1 Ship it!!

**Comment by smarthi** [comment]
We have to change the package names for test classes from present 'test.xx' to 'org.apache.pirk.xx'. That would be a different jira.

**Comment by ellisonanne** [comment]
Good catch Suneel!

**Pull request closed without merging**